### PR TITLE
feat(PTP-1472): sdk store + ipfs auth

### PR DESCRIPTION
**Pull request by salimtb**

### CODE REVIEW

File: src/lib/Auth/Auth.js [hunk 1]
```diff
@@ -71,12 +75,35 @@ export default class Auth {
...
     }
-
     this.#privateKey = privateKey;
     this.#projectId = projectId;
     this.#secretId = secretId;
     this.#chainId = chainId;
     this.#rpcUrl = rpcUrl;
+    this.#ipfs = ipfs;
+
+    if (ipfs) {
```
Comment (danielpavelicconsensys) [inline src/lib/Auth/Auth.js:85]: merge this pls

File: src/lib/Auth/Auth.js [hunk 2]
```diff
@@ -71,12 +75,35 @@ export default class Auth {
...
+        throw new Error(
+          errorLogger({
+            location: ERROR_LOG.location.Auth_constructor,
+            message: ERROR_LOG.message.no_ipfs_secretId_supplied,
+          }),
+        );
+      }
+
+      const { projectId: ipfsProjectId, apiKeySecret } = ipfs;
+      this.#ipfs = new IPFS({ projectId: ipfsProjectId, projectSecret: apiKeySecret });
```
Comment (danielpavelicconsensys) [inline src/lib/Auth/Auth.js:105]: project id and api key secret - sync it with infura pls

File: src/lib/SDK/sdk.js [hunk 3]
```diff
@@ -232,4 +237,117 @@ export default class SDK {
...
+      );
+    }
+    return this.#ipfsClient.uploadContent({ source: metadata });
+  }
+
+  /** Store data in ipfs
+   * @param {Array<any>} metadata an array of valid JSON Metadata
+   * @returns {Promise<string>} Ipfs hash of the stored data
+   */
+  async storeBaseURI(metadata) {
```
Comment (danielpavelicconsensys) [inline src/lib/SDK/sdk.js:296]: I don't like this storeBaseURI it's confusing rather storeContractMetadata?

File: src/lib/SDK/sdk.js [hunk 4]
```diff
@@ -232,4 +237,117 @@ export default class SDK {
...
+    }
+
+    return this.#ipfsClient.uploadArray({ sources: metadata });
+  }
+
+  /** Store data in ipfs
+   * @param {string} metadata any string
+   * @returns {Promise<string>} Ipfs hash of the stored data
+   */
+  async storeContent(metadata) {
```
Comment (danielpavelicconsensys) [inline src/lib/SDK/sdk.js:332]: can we rename to storeTokenMetadata or similar where is clear it's token related?

File: src/lib/SDK/sdk.js (hunk 3 above)
Comment (efecarranza) [reply src/lib/SDK/sdk.js:296]: `createFolder` might make more sense on what is actually being done

File: src/lib/SDK/sdk.js (hunk 4 above)
Comment (efecarranza) [reply src/lib/SDK/sdk.js:332]: `storeContent` and `storeMetadata` do the same thing basically...at the very least we can DRY up the code

**Review by javiergarciavera — APPROVED** [review]

**Review by danielpavelicconsensys — APPROVED** [review]

**Pull request merged**